src/Symfony/Component/Serializer/Normalizer/DataUriNormalizer.php

```diff
@@ -34,26 +32,14 @@ class DataUriNormalizer implements NormalizerInterface, DenormalizerInterface, C
     ];
 
     /**
-     * @var MimeTypeGuesserInterface
+     * @var MimeTypeGuesserInterface|null
      */
     private $mimeTypeGuesser;
 
-    /**
-     * @param MimeTypeGuesserInterface
-     */
-    public function __construct($mimeTypeGuesser = null)
+    public function __construct(MimeTypeGuesserInterface $mimeTypeGuesser = null)
     {
-        if ($mimeTypeGuesser instanceof DeprecatedMimeTypeGuesserInterface) {
-            @trigger_error(sprintf('Passing a %s to "%s()" is deprecated since Symfony 4.3, pass a "%s" instead.', DeprecatedMimeTypeGuesserInterface::class, __METHOD__, MimeTypeGuesserInterface::class), E_USER_DEPRECATED);
-        } elseif (null === $mimeTypeGuesser) {
-            if (class_exists(MimeTypes::class)) {
-                $mimeTypeGuesser = MimeTypes::getDefault();
-            } else {
-                @trigger_error(sprintf('Passing null to "%s()" without symfony/mime installed is deprecated since Symfony 4.3, install symfony/mime.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
this should be an exception now?

Reply:
That's what I thought too (see second commit), but the data uri normalizer used to fallback to `application/octet-stream` if no default mime type guesser can be instantiated. So neither was the http-foundation nor the mime type component a strict dependency to use it.

Reply:
this notice is wrong in the first place and should be removed from 4.3 then, right?

Reply:
Indeed. See https://github.com/symfony/symfony/pull/31923